provider/equinix/environ_network.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package equinix
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/names/v4"
+)
+
+var _ environs.Networking = (*environ)(nil)
+
+// Subnets returns basic information about subnets known by the provider for
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+	projectID := e.cloud.Credential.Attributes()["project-id"]
```

Comment:
Please add a check that `Attributes` is not nil before indexing the map.
Also please add a comment that we checked the presence of `project-id` when we were verifying the credentials.